atomicapp/nulecule/base.py

```diff
@@ -192,21 +192,20 @@ def load(self, nodeps=False, dryrun=False):
             else:
                 self.load_external_application(dryrun)
 
-    def run(self, provider_key, dry=False):
+    def run(self, provider_key, dryrun=False):
         if self._app:
-            self._app.run(provider_key, dry)
+            self._app.run(provider_key, dryrun)
             return
-        provider_key, provider = self.get_provider(provider_key, dry)
+        provider_key, provider = self.get_provider(provider_key, dryrun)
         provider.artifacts = self.rendered_artifacts.get(provider_key, [])
         provider.init()
-        if not dry:
-            provider.deploy()
+        provider.deploy()
 
     def stop(self, provider_key=None, dryrun=False):
         if self._app:
-            self._app.stop(provider_key)
+            self._app.stop(provider_key, dryrun)
             return
-        provider_key, provider = self.get_provider(provider_key)
+        provider_key, provider = self.get_provider(provider_key, dryrun)
         provider.artifacts = self.rendered_artifacts.get(provider_key, [])
         provider.init()
         provider.undeploy()
```

Comment:
this is :+1: but we didn't pass dryrun at all in the first place :) it's all passing now.